tests/Unit/Domain/Creators/Test_CylindricalBinaryCompactObject.cpp

```diff
@@ -600,16 +538,126 @@ void test_parse_errors() {
+    CAPTURE(with_sphere_e);
+    CAPTURE(include_outer_sphere);
+    CAPTURE(use_equiangular_map);
+    CAPTURE(with_boundary_conditions);
+    CAPTURE(with_additional_outer_radial_refinement);
+    CAPTURE(with_additional_grid_points);
+    CAPTURE(with_time_dependence);
+    bool with_control_systems = false;
+    if (with_time_dependence) {
+      with_control_systems = gsl::at(
+          random_sample<1>(make_array(true, false), make_not_null(&gen)), 0);
+      // For time dependent maps we must include the inner spheres
+      include_inner_sphere_A = true;
+      include_inner_sphere_B = true;
+    }
+    CAPTURE(include_inner_sphere_A);
+    CAPTURE(include_inner_sphere_B);
+    CAPTURE(with_control_systems);
+
+    const double outer_radius = include_outer_sphere ? 100.0 : 30.0;
+    const double mass_ratio = with_sphere_e ? 4 : 1.2;
+    // Set centers so that the Newtonian COM is at the origin,
+    // except offset slightly in the y direction.
+    const std::array<double, 3> center_objectA = {
+        separation / (1.0 + mass_ratio), y_offset, 0.0};
+    const std::array<double, 3> center_objectB = {
+        -separation * mass_ratio / (1.0 + mass_ratio), y_offset, 0.0};
+
+    // Set radius = 2m for each object.
+    // Formula comes from assuming m_A+m_B=1.
+    const double inner_radius_objectA = 2.0 * mass_ratio / (1.0 + mass_ratio);
+    const double inner_radius_objectB = 2.0 / (1.0 + mass_ratio);
+
+    std::unique_ptr<DomainCreator<3>> cyl_binary_compact_object{};
+    if (with_time_dependence) {
+      if (with_boundary_conditions) {
+        cyl_binary_compact_object =
+            TestHelpers::test_option_tag<domain::OptionTags::DomainCreator<3>,
+                                         Metavariables<3, true, true>>(
+                create_option_string(
+                    with_time_dependence,
+                    with_additional_outer_radial_refinement,
+                    with_additional_grid_points, include_outer_sphere,
+                    with_boundary_conditions, use_equiangular_map,
+                    center_objectA, center_objectB, inner_radius_objectA,
+                    inner_radius_objectB, outer_radius));
+      } else {
+        cyl_binary_compact_object =
+            TestHelpers::test_option_tag<domain::OptionTags::DomainCreator<3>,
+                                         Metavariables<3, true, false>>(
+                create_option_string(
+                    with_time_dependence,
+                    with_additional_outer_radial_refinement,
+                    with_additional_grid_points, include_outer_sphere,
+                    with_boundary_conditions, use_equiangular_map,
+                    center_objectA, center_objectB, inner_radius_objectA,
+                    inner_radius_objectB, outer_radius));
+      }
+
+      const auto& [initial_expiration_times, expected_functions_of_time] =
+          construct_expected_functions_of_time(with_control_systems);
```

Comment:
Does this actually test something useful? It seems to me that it mostly reimplements the FoT function of the domain creator. I think it would be best to delete this test of the functions of time (which also gets rid of the `if (with_time_dependence)` clause) and instead check in `test_construction` that a few specific coordinates move around as you would expect. There's a super simple example in `Test_Interval.cpp`. We can defer this to a follow-up PR as well.

Reply:
For the BCO domains where we manually enter the time dependent maps, I think something like this is useful, but yes please lets defer this until later

Reply:
ok